src/resources/projects.js

```diff
@@ -0,0 +1,124 @@
+import express from 'express';
+import fs from 'fs';
+
+const router = express.Router();
+const projects = require('../data/projects.json');
+
+router.get('/get', (req, res) => {
+  res.send(projects);
+});
+
+router.get('/get/:id', (req, res) => {
+  const projectId = parseInt(req.params.id, 10);
+  const foundProject = projects.find((project) => project.id === projectId);
+  if (foundProject) {
+    res.send(foundProject);
+  } else {
+    res
+      .status(400)
+      .json({ msg: `There is no project with the id of ${req.params.id}` });
+  }
+});
+
+router.get('/getByName/:name', (req, res) => {
+  const projectName = req.params.name;
+  const foundProject = projects.find((project) => project.projectName === projectName);
+  if (foundProject) {
+    res.send(foundProject);
+  } else {
+    res
+      .status(400)
+      .json({ msg: `There is no project with the name of: ${req.params.name}` });
+  }
+});
+
+router.get('/getByDate/:date', (req, res) => {
+  const projectStartDate = req.params.date;
+  const foundProject = projects.find((project) => project.startDate === projectStartDate);
+  if (foundProject) {
+    res.send(foundProject);
+  } else {
+    res
+      .status(400)
+      .json({ msg: `There is no project iniciated on ${req.params.date}. The date format must be (dd-mm-yyyy)` });
+  }
+});
+
+router.post('/add', (req, res) => {
+  const bodys = req.body;
+  const idNewProject = new Date().getTime().toString().substring(6);
+  const newEmployees = [];
+  const newTasks = [];
+  bodys.employees.forEach((employee) => {
+    newEmployees.push({
+      id: employee.id,
+      name: employee.name,
+      rol: employee.rol,
+      rate: employee.rate,
+    });
+  });
+
+  bodys.tasks.forEach((task) => {
+    newTasks.push( {
```

Comment:
Be carefull with the use of spaces
```suggestion
    newTasks.push({
```